### ISSUE TITLE
SIGSEGV on start

### ISSUE THREAD

**Issue by sonicpp** [eringerli/QtOpenGuidance#1]
Hi,
I finally compilled your project (commit f4fc5b71b97350dcd4143bfe7f5c3fec36828d9c) and I invoke SIGSEGV every time I try to run it.

```
./QtOpenGuidance
DPI:  96 96 344 194
SPNAV_ENABLED
connect failed: No such file or directory
maximumAnisotropy 16
QObject::connect: No such slot GlobalPlanner::setPassNumber( int )
QObject::connect:  (sender name:   'PassToolbar')
```
Profile: Release
Qt version: 5.14
Distro: Arch Linux

**Comment by eringerli** [comment]
This one is caused by QT 5.14, updating to 5.15 (or down to 5.13, but not tested) should resolve it. I somehow hit a sweet spot in QT3D 🤷‍♂️

**Comment by sonicpp** [comment]
Thanks for info. Once the Qt 5.15 gets out of beta and get it into Arch Linux, I will test it and send my reports back.

P.S.: Hmm, looks like Qt 5.15 release will not come before May...

**Comment by eringerli** [comment]
I have installed 5.15 it in Arch Linux too, but it takes a couple of GBs. Better they take their time and do it right

**Comment by sonicpp** [comment]
Well, if you have Arch too and it is working on 5.15, then feel free to close it.

**Comment by sonicpp** [comment]
I got some upgrades on Arch

- qt5-base (5.14.1-1 -> 5.14.1-2 -> 5.14.1-3)
- qt5-3d (5.14.1-1 -> 5.14.1-2)

And now it runs without a crash! It was probably a gcc error, see [qt5-3d package](https://git.archlinux.org/svntogit/packages.git/commit/trunk?h=packages/qt5-3d&id=efaaa5f1d4255e6b490b9b8bfa28c0e319581aa7)

The 3D window is grey and empty, but that is another story

**Comment by eringerli** [comment]
I confirmed it, it runs on stock Arch Linux without problems. Installed version is 5.14.1-2.

You have to load a config (`minimal.json` is kept up to date with the development), as the camera is tied to the calculated position. Also all objects in the view are configured with the blocks in the settings dialog and you have to enable the simulator to get the calculations going. If this sounds complicated, you get the system instantly once you loaded the config.

**Comment by sonicpp** [comment]
Now I get it. Thanks :)

**Comment by eringerli** [comment]
if you want, you can write me on telegram on @eringerli, so I can add you to the group for discussing stuff and keeping up on the progress.

**Comment by eringerli** [comment]
a new bugreport about this issue, but this time for manjaro on the RPI4: https://forum.manjaro.org/t/compile-qt5-3d-for-es2-too-bug-in-qt5-3d/133958, and in turn on Arch Linux Arm too: https://archlinuxarm.org/forum/viewtopic.php?f=15&t=14396

**Comment by eringerli** [comment]
also for MSYS2 on windows: https://github.com/msys2/MINGW-packages/issues/6298

**Comment by eringerli** [comment]
reopen as it still affects Manjaro ARM on the RPI4: its toolchain has still GCC 9.2.0, Manjaro on x86_64 has GCC 9.3.0

**Comment by eringerli** [comment]
MSYS2 works now, but I'm still waiting on the Arch Linux Arm team to update their toolchain

**Comment by eringerli** [comment]
it is updated, so it works now without compiling issues
